stage1/appexec/appexec.c

```diff
@@ -256,6 +260,16 @@ static void parse_gids(const char *str, size_t *n_gids_p, gid_t **gids_p)
 
 int main(int argc, char *argv[])
 {
+	char *term = getenv("TERM"); /* usefull to keep during entering. */
+	int c;
+
+	while ((c = getopt(argc, argv, "e")) != -1)
+		switch (c) {
+			case 'e':
+				setenv("TERM", term, 1);
```

Comment:
you forget to move argv offset according the number of options from `optind` and when you do this (with handling of '--'), you can as well handle `-e NAME_OF_VARIABLE` to fill the keepEnv table with variables that should be passed:
so it at the end:
- `run` -> unit.service -> ExecStart = `appexec -e LISTEN_FDS -e LISTEN_PID ...`
- 'enter' -> `appexec -e TERM`

just an idea, but look as better, more clean, and more extensible solution - only if you have time for this :smile_cat: